### ISSUE TITLE
Cache resolved lobby host address

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#4907]
Task: cache the resolved host address for the lobby and use it as a fallback in case DNS is not working.

This would have prevented this issue from having been a major problem: https://github.com/triplea-game/triplea/issues/4904

Today we cache the lobby host address and use that as a fallback if github is not available to fetch lobby-properties, this task would have us go one step further and do additional caching.

**Comment by DanVanAtta** [comment]
@RoiEXLab as an idea, what if we had the IPv6 address of lobby instead of the DNS address? In that case we would not need to cache the resolved address and we would have no problem with IPv6 only plans. AFAIK all java versions going forward will have no problem with IPv6 (but I could be wrong or missing something maybe). Curious what you think.

**Comment by RoiEXLab** [comment]
Unfortunately not all ISPs offer IPv6, so there might be some users that can't use that address :(

**Comment by DanVanAtta** [comment]
@RoiEXLab thought of one solution for this, place both IPv4 and IPv6 address in the config file rather than DNS name. thoughts?

**Comment by RoiEXLab** [comment]
I mean we could have all three variants listed (DNS, IPv6, IPv4) and just try them in this order until something works.

The thing with DNS is that it offers flexibility: We could even use DNS to load balance servers if the lobby would support that.

But in the end having 2 or more ip addresses in this file seems like we're essentially creating a small DNS-like service ourselves.
I mean it was bad when the DNS service didn't work as expected, but at the same time;
 - how high are the odds for this happening again in the near future?
- In case GitHub is unavailable (which is of course unlikelier than our case, but the cloadflare thing happened, so you never know) our approach wouldn't work either,  whereas Hardcoding a DNS into the engine doesn't really reduce the amount of flexibility but keep github out of everything (in theory).

TL;DR: In my opinion it's OK relying on DNS. There are so many things we depend on working  that DNS is probably just one of many reasons why the lobby is down. Also in case we want to encrypt the connection at some point we'd have to go with dns-only anyways, because certificates only are issued for domains, not IPs

**Comment by DanVanAtta** [comment]
Agree with your points on benefit of DNS @RoiEXLab ; though I don't think we get any of those benefits just yet. The HTTPS server would want/need a DNS name, so you're right we'd probably need all 3.

Since the game engine caches the resolved address, we'd be 'immune' to any short-lived DNS problems. Grant it, DNS is fundamental and rarely has issues, but because something rarely happens is not good argument for not handling it.

I suspect the thing to do would be to use DNS, and in-code resolve it and cache the resolved address.

The HTTPS server piece is interesting. I'd be a bit surprised if we can't update something to tell java, "here is a net address, and it resolves to this IP, use it! And skip the DNS lookup!" 

The ROI of this effort I think does become questionable. I'll ice-box this issue in light of that consideration.
